Engine.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Engine.h"
+using namespace sf;
+
+//Public functions only here (private functions in Input.cpp, Update.cpp and Draw.cpp respectively)
+
+
+Engine::Engine()
+{
+	
+
+	//Get the screen resolution and create an SFML window and fullscreen view
+	Vector2f resolution;
+	resolution.x = VideoMode::getDesktopMode().width;
+	resolution.y = VideoMode::getDesktopMode().height;
+
+	m_Window.create(VideoMode(resolution.x, resolution.y), "Catch all the Cupcakes game!!!", Style::Fullscreen);
+	//m_Window.setFramerateLimit(60);
+	
+	//Associate the sprite with the texture and set the position to render
+	m_BackgroundSprite = Sprite(TextureHolder::GetTexture("Images/background.png"));
+	m_BackgroundSprite.setPosition(0, 0);
+
+	
+	
+}
+
+void Engine::run()
+{
+	//Timing
+	Clock clock;
+
+
+	while (m_Window.isOpen())
+	{
+		Time dt = clock.restart();
+		//Update the total game time
+		m_GameTimeTotal += dt;
+		
+				//Call each part of the game loop in turn
+				input();
+				
+				draw();
+	}
+
+		
+	
+}
+
+
+
+

```

Comment:
Okay, come to think of it, regarding spaces and stuff. It is in the end of the day your style! If you like it this way, disregard my comments! But one thing I still recommend, keep it consistent, that makes it easier to read (I am commenting this now because I am slowly getting used to the space and I realized I have been preaching without need 😄).